deps/rabbitmq_shovel/src/rabbit_amqp091_shovel.erl

```diff
@@ -230,6 +230,9 @@ handle_source({'EXIT', Conn, Reason},
               #{source := #{current := {Conn, _, _}}}) ->
     {stop, {inbound_conn_died, Reason}};
 
+handle_source({'EXIT', _Pid, {shutdown, {server_initiated_close, ?PRECONDITION_FAILED, Reason}}}, _State) ->
+    {stop, {inbound_consumer_no_ack, Reason}};
```

Comment:
This return value of `inbound_consumer_no_ack` is not suitable for every `406 PRECONDITION FAILED` response. We should change it to `source_precondition_failed`